File: Javascript/Misc Learning/exercism day 1/newtons.hs
squareRoot :: Double -> Int
squareRoot radicand
    | radicand < 0 = error "Radicand must be a non-negative number"
    | otherwise    = squareRoot' (toInt (radicand * 100)) (toInt ((radicand * 100) / 2))
    where
        toInt :: Double -> Int
        toInt x = round x `shiftR` 7  -- 10^7 is a large enough power of 10 for our purposes
        
        squareRoot' :: Int -> Int -> Int
        squareRoot' rad guess
            | abs (guess * guess - rad) <= 100 = guess
            | otherwise = squareRoot' rad ((guess + rad `div` guess) `div` 2)

File: Javascript/Misc Learning/exercism day 1/newtons.js
export const squareRoot = (radicand) => {
    if (radicand < 0) {
        throw new Error("Radicand must be a non-negative number");}

    let guess = radicand / 2;     // Initial guess for the square root
    while (Math.abs(guess * guess - radicand) > .1) { // Iterate until the guess is close enough
        guess = (guess + radicand / guess) / 2;
        guess = guess >> 0}
    return guess;};

module.exports = squareRoot;


//    const squareRoot = require('./squareRoot');
